### DISCUSSION TITLE
typed osd when change zoom level from 20x to 40x the same image becomes zoomed yet blurred

### DISCUSSION THREAD

**Discussion by stevendai11** [openseadragon/openseadragon#2048] [Q&A]
Hi,
I am using typed osd (version 2.4.1). As the title suggested, the image is zoomed but blurred when changing zoom level from 20x to 40x for example.
I tried the parameter like 
(options as any).immediateRender = true;
 options.imageSmoothingEnabled = false;
 but without much luck.
No sure if there is any improvement in the latest osd like version 2.4.8?
Thanks a lot for your help.
Steven

**Comment by iangilman** [comment]
Does your image have enough resolution to be displayed at 40x? 

You can do `viewer.viewport.viewportToImageZoom(viewer.viewport.getZoom())` to find out how close you are to the image's native resolution at that point. A 1 means you are at 1:1, and a 2 means you are zoomed in twice the native resolution, etc. 

BTW, turning off `imageSmoothingEnabled` should make the image more boxy, but it can't make up detail that isn't in the image.

**Reply by stevendai11** [reply]
Thanks a lot Ian.
You are right my image is scanned at 20x and viewed at 40x so it causes problem. I will scan it at 40x to see if it make difference.
BTW, I have upgraded to osd 2.4.2 but not much difference.
Also I have an additional question if I could. 
For a large image, if I click on an area in FOV or navigator windows which is far from current center (viewport), the switch causes the redraw of the viewport with updated image (or tiles) which is correct. But somehow some tiles are updated faster than others in the same viewport area. When I set ImmediateRender to true, you can see the area is updated tile by tile or square by square (since the background is white) and if I set the ImmediateRender to false, you wouldn't see the tile update , but you can see some area is blurred for a while since it is redrawing on top of the old tile(s).
Is there a way to make the caching large enough to cache all the tiles in the image so moving to a different area for which tiles could always be cached?
Thanks a lot again.
Steven

**Reply by iangilman** [reply]
Well, the point of OpenSeadragon is not to load the entire image, since the image could be much larger than you want to transfer or hold in memory at once. You could potentially make the cache big enough to hold the whole image (if the image was small enough, or if the user had enough memory), but even then we don't have any facility to pre-load tiles that are outside of the current viewing area. 

In practice, none of this is an issue, as long as your server is able to deliver the tiles in a timely manner. I imagine you are noticing the blurred areas because it's taking a while to load your tiles (as discussed in #2045).

**Reply by stevendai11** [reply]
Thanks.
Actually I loaded whole image at the beginning so there is no server reload during clicking on a different FOV area. It's up to osd to decide how to cache the whole image, let's say 2G in total. There is only one image at a time in osd.

**Reply by iangilman** [reply]
Fair enough. Do you know how many tiles are in your image? You can try setting the `maxImageCacheCount` option to that or a bigger number. 

Who is your audience for this? Most people aren't going to be able to download 2GB in a "reasonable" amount of time, I think.
